### PR TITLE
Generator Setup

### PR DESCRIPTION
**Pull request by HectorMg**
What does this PR do?

* Creates a jsmodels subdir where the generator will live
* Adds the commander module to enable command line application development
* Implements the basic `jsmodels generate <ModelName>` command.
* Adds README.md with basic info on the `generate` command

### CODE REVIEW

File: jsmodels/index.js [hunk 1]
```diff
@@ -1,118 +1,30 @@
...
-    let modelParams = {}
-    this.constructor.ATTRIBUTES().forEach((attribute) => {
-      modelParams[attribute] = this[attribute]
-    })
-
-    console.log("Retrieven model params: " + JSON.stringify(modelParams))
-    return modelParams
-  }
+#!/usr/bin/env node
+'use strict';
```
Comment (walreyes) [inline jsmodels/index.js]: Move this to it's own folder `/generator/index.js`. So you need to run `npm link` inside the `generator` folder

File: jsmodels/index.js [hunk 2]
```diff
@@ -1,118 +1,30 @@
...
+const program = require("commander"),
+  exec = require('child_process').exec;
+
+let generateFiles = (name,attributeList) => {
+  let execCallback = (error, stdout, stderr) => {
+     if (error) console.log("exec error: " + error);
+     if (stdout) console.log("Result: " + stdout);
+     if (stderr) console.log("shell error: " + stderr);
+   };
+  const cmd = "touch ";
```
Comment (walreyes) [inline jsmodels/index.js]: You are just creating a new empty file, remember that we talked about the template.

File: jsmodels/package.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "jsmodels",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "dependencies": {
+    "commander": "^2.11.0"
+  },
+  "bin": {
+    "jsmodels": "./index.js"
```
Comment (walreyes) [inline jsmodels/package.json]: I guess that it would need to be something like:

` "jsmodels": "./generator/index.js"`

**Comment by HectorMg** [comment]
@walreyes ya quedó refactoreado, solo faltan las specs

**Comment by walreyes** [comment]
@HectorMg I'll merge this to my branch.

You can checkout a branch from `feature/jsmodels` for the refactor to a class and add the specs.

**Pull request merged**